src/browser/BrowserMonitoringService.ts

```diff
@@ -0,0 +1,78 @@
+import { datadogLogs } from '@datadog/browser-logs';
+
+import { Logger } from '../shared/Logger';
+import { ConsoleLogger } from '../shared/ConsoleLogger';
+
+const ddLogger = datadogLogs.logger;
+
+interface RemoteMonitoringServiceParams {
+  serviceName: string;
+  serviceVersion: string;
+  serviceEnv: string;
+  clientToken: string;
+}
+
+export class BrowserMonitoringService {
+  private logger: Logger;
+  private remoteMonitoringServiceParams:
+    | RemoteMonitoringServiceParams
+    | undefined;
+
+  constructor(remoteMonitoringServiceParams?: RemoteMonitoringServiceParams) {
+    const { serviceName, serviceVersion, serviceEnv, clientToken } =
+      remoteMonitoringServiceParams ?? {};
+
+    const isSomeRemoteMonitoringServiceParamMissing =
```

Comment:
RemoteMonitoringServiceParams type obliges you to provide all params 🤔

Reply:
This is correct. DataDog won't work without any of these I assume. Maybe it can work without version or env, but I don't think we want it to work without tit - since it can make logs messy. WDYT?